pkg/catalog/manager/crd.go

```diff
@@ -52,7 +56,10 @@ func (m *Manager) getTemplateMap(catalogName string, namespace string) (map[stri
 }
 
 func (m *Manager) updateTemplate(template *v3.CatalogTemplate, toUpdate v3.CatalogTemplate) error {
-	r, _ := labels.NewRequirement(TemplateNameLabel, selection.Equals, []string{template.Name})
+	r, err := labels.NewRequirement(TemplateNameLabel, selection.Equals, []string{template.Name})
+	if err != nil {
```

Comment:
same thing here with adding to the error, I don't think it'll make a lot of sense to a user, maybe something to indicate that it's probably not an issue with their catalog. Have you seen an error here at all?

Reply:
if a template's template version failed to create, but the template successfully created this causes  r to be nill during update because templatenamelabel doesn't exist on that template. Then the next call `m.templateVersionLister.List(template.Namespace, labels.NewSelector().Add(*r))` was being called with a nil r and panicking.